.github/workflows/tests-with-lower.yml

```diff
@@ -0,0 +1,85 @@
+name: Tests with the oldest dependencies
```

Comment:
The difference between this file and `tests.yaml` seems to be limited:

```diff
1c1
< name: Tests
---
> name: Tests with the oldest dependencies
51c51
<     - name: Install
---
>     - name: Setup pip
55a56,63
>     - name: Install dependencies with lower bound constraints
>       run: |
>         # Install dependencies with lower bound constraints.
>         pip install alembic==1.5.0 cmaes==0.8.2 packaging==20.0 sqlalchemy==1.3.0
>         pip install plotly==4.0.0 scikit-learn==0.24.2  # optional extras
> 
>     - name: Install
>       run: |
```

It may be costly to maintain both files. Can we merge these files?
I'm not sure, but we may be able to use `matrix` and `if` in `tests.yaml` as follows:


```diff
diff --git a/.github/workflows/tests.yml b/.github/workflows/tests.yml
index 00d346cd3..1b29b7d4d 100644
--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -19,6 +19,7 @@ jobs:
     strategy:
       matrix:
         python-version: ['3.6', '3.7', '3.8', '3.9', '3.10']
+        lower-bound: [true, false]
 
     services:
       redis:
@@ -48,11 +49,20 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-${{ matrix.python-version }}-${{ env.cache-name }}-${{ hashFiles('**/setup.py') }}
 
-    - name: Install
+    - name: Setup pip
       run: |
         python -m pip install --upgrade pip
         pip install --progress-bar off -U setuptools
 
+    - name: Install dependencies with lower bound constraints
+      run: |
+        # Install dependencies with lower bound constraints.
+        pip install alembic==1.5.0 cmaes==0.8.2 packaging==20.0 sqlalchemy==1.3.0
+        pip install plotly==4.0.0 scikit-learn==0.24.2  # optional extras
+      if: ${{ matrix.lower-version }}
+
+    - name: Install
+      run: |
         # Install minimal dependencies and confirm that `import optuna` is successful.
         pip install --progress-bar off .
         python -c 'import optuna'
```

Reply:
Thank you so much for the advice, it really makes sense. I updated the PR based on your suggestion.